Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM python:3.7.3-slim as base
+
+# change /src/app to /airflow
+ENV AIRFLOW_HOME=/airflow
+ENV AIRFLOW__CORE__DAGS_FOLDER=${AIRFLOW_HOME}/dags
+ENV AIRFLOW__CORE__BASE_LOG_FOLDER=${AIRFLOW_HOME}/logs
+ENV AIRFLOW__CORE__PLUGINS_FOLDER=${AIRFLOW_HOME}/plugins
+ENV AIRFLOW__CORE__CHILD_PROCESS_LOG_DIRECTORY=${AIRFLOW_HOME}/logs/scheduler
+
+ENV AIRFLOW__CORE__SQL_ALCHEMY_CONN=postgresql+psycopg2://postgres:example@db/airflow-db
+ENV AIRFLOW__CORE__LOAD_EXAMPLES=false
+
+WORKDIR ${AIRFLOW_HOME}
+
+COPY ./requirements ./requirements
+
+RUN pip install -U pip \
+    && set -ex \
+    && apt-get update -yq \
+    && apt-get install -yq --no-install-recommends build-essential \
+    && pip install --no-cache-dir -r ./requirements/base.txt \
+    && apt-get remove --purge --autoremove -yq build-essential \
+    && rm -rf ~/.cache/* \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /tmp/*
+
+FROM base as dev
+
+ENV PYTHONUNBUFFERED=true
+ENV PYTHONDONTWRITEBYTECODE=true
+
+RUN pip install --no-cache-dir -r ./requirements/dev.txt \
+    && rm -rf ~/.cache/* \
+    && rm -rf /tmp/*
+
+ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.5.0/wait /wait
+RUN chmod +x /wait
+
+EXPOSE 8080
+
+CMD /wait && airflow initdb && airflow webserver -p 8080 && airflow scheduler
```

Comment:
would check if `airflow webserver` is asynchronous and returns? Otherwise `airflow scheduler` wouldn't get executed. If the intention is to have to separate processes they are usually implemented with two containers running the same image

Reply:
I'm unsure of the implementation but this is what the `airflow` documentation says to do which I think is fine for development purposes, especially for a prototype.

Reply:
This command doesn't seem to be running `airflow scheduler`:
```
 3863 ?        Sl     0:00  \_ containerd-shim -namespace moby -workdir /var/lib/containerd/io.containerd.runtime.v1.linux/moby/808de0303907108b43b141a4fc8ce1aa55a85437f05c18db2462d80ab752da42 -address /run/containerd/containerd.sock -containerd-binary /usr/bin/containerd -runtime-root /var/run/docker/runtime-runc
 3897 ?        Ss     0:00      \_ /bin/sh -c /wait && airflow initdb && airflow webserver -p 8080 && airflow scheduler
 4199 ?        S      0:01          \_ /usr/local/bin/python /usr/local/bin/airflow webserver -p 8080
 4275 ?        S      0:01              \_ gunicorn: master [airflow-webserver]
 4287 ?        Sl     0:00                  \_ [ready] gunicorn: worker [airflow-webserver]
 4290 ?        Sl     0:00                  \_ [ready] gunicorn: worker [airflow-webserver]
 4292 ?        Sl     0:00                  \_ [ready] gunicorn: worker [airflow-webserver]
 4540 ?        Sl     0:00                  \_ [ready] gunicorn: worker [airflow-webserver]
```
because `airflow-webserver -p 8080` (correctly) blocks while it is running.

Reply:
correct, I've just seen there is a `/health` endpoint indicating this. Giving the official documentation another read.